src/bp/core/repositories/users.ts

```diff
@@ -62,15 +65,29 @@ export class KnexUserRepository implements UserRepository {
     return { result: newUser, created: true }
   }
 
-  async updateAttributes(channel: string, id: string, attributes: any): Promise<void> {
+  async getAttributes(channel: string, user_id: string): Promise<any> {
+    const user = await this.database
+      .knex(this.tableName)
+      .where({ channel, user_id })
+      .limit(1)
+      .select('attributes')
+      .first()
+
+    return this.database.knex.json.get(user.attributes)
+  }
+
+  async updateAttributes(channel: string, user_id: string, attributes: any): Promise<void> {
     channel = channel.toLowerCase()
 
+    if (await this.dataRetentionService.hasPolicy()) {
+      const originalAttributes = await this.getAttributes(channel, user_id)
+      await this.dataRetentionService.checkChanges(channel, user_id, originalAttributes, attributes)
```

Comment:
normally I would expect `check` to only do a check, i.e. have no side effect
so this method seemed a bit counterintuitive since it doesn't return anything